packages/slate-editor/src/components/EditorBlock/EditorBlock.module.scss

```diff
@@ -4,11 +4,6 @@
 .block {
     @include editor-v4-block($void: false);
     position: relative;
-    white-space: normal;
```

Comment:
This was causing the problem. 

And also setting `user-select: none` for `.void` elements too.

Big kudos to @aspirisen for helping finding it. :trophy: